omnigibson/robots/manipulation_robot.py

```diff
@@ -802,10 +808,74 @@ def _calculate_in_hand_object_rigid(self, arm="default"):
+        assert (
+            self.assisted_grasp_start_points[arm] is not None
+        ), "In order to use assisted grasping, assisted_grasp_start_points must not be None!"
+        assert (
+            self.assisted_grasp_end_points[arm] is not None
+        ), "In order to use assisted grasping, assisted_grasp_end_points must not be None!"
+
+        # Iterate over all start and end grasp points and calculate their x,y,z positions in the world frame
+        # (per arm appendage)
+        # Since we'll be calculating the cartesian cross product between start and end points, we stack the start points
+        # by the number of end points and repeat the individual elements of the end points by the number of start points
+        startpoints = []
+        endpoints = []
+        for grasp_start_point in self.assisted_grasp_start_points[arm]:
+            # Get world coordinates of link base frame
+            link_pos, link_orn = self.links[grasp_start_point.link_name].get_position_orientation()
+            # Calculate grasp start point in world frame and add to startpoints
+            start_point, _ = T.pose_transform(link_pos, link_orn, grasp_start_point.position, [0, 0, 0, 1])
+            startpoints.append(start_point)
+        # Repeat for end points
+        for grasp_end_point in self.assisted_grasp_end_points[arm]:
+            # Get world coordinates of link base frame
+            link_pos, link_orn = self.links[grasp_end_point.link_name].get_position_orientation()
+            # Calculate grasp start point in world frame and add to endpoints
+            end_point, _ = T.pose_transform(link_pos, link_orn, grasp_end_point.position, [0, 0, 0, 1])
+            endpoints.append(end_point)
+        # Stack the start points and repeat the end points, and add these values to the raycast dicts
+        n_startpoints, n_endpoints = len(startpoints), len(endpoints)
+        raycast_startpoints = startpoints * n_endpoints
+        raycast_endpoints = []
+        for endpoint in endpoints:
+            raycast_endpoints += [endpoint] * n_startpoints
+
+        # Calculate raycasts from each start point to end point -- this is n_startpoints * n_endpoints total rays
+        ray_data = set()
+        # Repeat twice, so that we avoid collisions with the fingers of each gripper themself
+        for raycast_startpoint, raycast_endpoint in zip(raycast_startpoints, raycast_endpoints):
+            origin = carb.Float3(raycast_startpoint)
+            dir = carb.Float3(raycast_endpoint - raycast_startpoint)
+            distance = np.linalg.norm(dir)
+            hit = get_physx_scene_query_interface().raycast_closest(origin, dir, distance)
+            if hit["hit"]:
+                # filter out self body parts (we currently assume that the robot cannot grasp itself)
+                if self.prim_path not in hit["rigidBody"]:
+                    ray_data.add(hit["rigidBody"])
```

Comment:
maybe consider using the util function like `from omnigibson.utils.sampling_utils import raytest_batch`?

Reply:
done